### ISSUE TITLE
dotnet-inspect

### ISSUE THREAD

**Issue by natemcmaster** [aspnet/DotNetTools#251]
A tool to inspect detailed information about the current setup. Here is a list of information that might be useful for support engineers, our team, and devs in general.

- optimized package cache. Available? In use? Cache misses? etc.
- Show graph of dependencies
  - what is lifting/downgrading specific package version
  - show sources of package conflicts
- list available shared runtimes (peek inside dotnet/sdk)

cc @glennc

**Comment by natemcmaster** [comment]
cc @muratg related to the diagnostic middleware we were dreaming of yesterday. Might be possible to code share between a CLI tool and a middleware.

**Comment by muratg** [comment]
This sounds like a great idea to me. Would help a lot with investigations.

@davidfowl @glennc  thoughts?

**Comment by glennc** [comment]
Yeah, we've talked about a tool like this from time to time. It sounds like a fine idea :)

**Comment by natemcmaster** [comment]
@muratg https://github.com/dotnet/cli/issues/5961#issuecomment-285806941

**Comment by natemcmaster** [comment]
More ideas:

 - sub-commands for gathering 
   - process dumps - http://blogs.microsoft.co.il/sasha/2017/02/26/analyzing-a-net-core-core-dump-on-linux/
   - performance data -  http://blogs.microsoft.co.il/sasha/2017/02/27/profiling-a-net-core-application-on-linux/, https://github.com/dotnet/coreclr/blob/master/Documentation/project-docs/linux-performance-tracing.

Currently, these profiling and process dumps are difficult on xplat. Improved command-line tooling would help.

**Comment by davidfowl** [comment]
YESSSSS

**Comment by muratg** [comment]
Putting this in the next milestone, pending other current Tooling discussions. We may end up trying to do it in 2.1 still.

**Comment by jzabroski** [comment]
@natemcmaster 
See also https://github.com/Microsoft/perfview/pull/732
It would be nice if the "steroids" command:
* Allowed Fusion Logging (does this exist in .NET Core like it does on .NET 461?)
* A GUI for automatically analyzing the performance data, like java has with [TDA (Thread Dump Analyzer)](http://java.net/projects/tda)

**Comment by muratg** [comment]
We don't have plans to add this in 2.2. Tentatively moving out to 3.0.0
